### ISSUE TITLE
[FEATURE] Keybinding help window takes a little long time to pop up. (Isn't blazingly fast/smooth)

### ISSUE THREAD

**Issue by arkochan** [mylinuxforwork/dotfiles#406]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when keybinding help window takes a long time (atleast more than a second) to pop up. This is due to the fact that each time `hypr/scripts/keybindings.sh` is called the whole .conf file is read to auto generate the `keybinds` variable which is later used as an input redirection to launch a rofi window for the list. 

**Describe the solution you'd like**
Rather than generating the `keybinds` variable from the keybind .conf (eg. `hypr/keybindings/default.conf`) file each time `hypr/scripts/keybindings.sh` script is ran, it can be generated once 
after each modification. It can be done in this process: 
1. check if a generated help file already exists for the selected keybind .conf file.
2. if not: generate a file from the `keybinds` variable.
3. else: use already generated help file

To ensure for each modification a single file is generated if can be checked either using the last modification using `stat -c '%Y' $config_file` or by generating a hash value for the file.  

**Additional context**
Only the while loop to read default.conf file in `hypr/scripts/keybindings.sh` takes about ~820 milliseconds on my machine. This is apart from the time taken by rofi to start after the 0.2s sleep. Which sums up to about a couple of seconds sometimes. Meanwhile there is no confirmation whether I've hit the right keybind for the keybind help or not.

**Comment by Stewart86** [comment]
Yes this makes sense.. It is more specifically for the distro-level concern than a compositor level one. However like many things, etc shell, terminal, browser? They are not required for the compositor to work properly? I don't know? In my mind I was just thinking that I only need `archinstall` and `ml4w-hyprland-setup` and I have fully rebuilt a system. In fact it is almost complete solution now, it's just missing a few things, one of which happened to be a good snapshot solution. If that was your initial intention to create this dotfiles, I think you should aim to create an installer that allow all hands off. Hack, you could even wrap `archinstall` and user would just need to call one line command and get everything up and running.

Yes, those utilities that I've mentioned might be more of necessities,but a robust snapshot backup is also a necessity for Arch based distro. Could be a rather gray area? Of cause if you say that you are trying to make this install script be compatible with other distributions then this will not be that important (still important imo), but you could also conditionally enable this feature based on distro?

ps: alternatively maybe the `linutil`by Chris Titus Tech might be a better solution.
